tests/unit/store/printMyBarcode/actions.spec.js

```diff
@@ -1,137 +1,81 @@
+        { ...printJobOptions },
+      )
 
-      const response = await Actions.printJob({ getters }, params)
+      const { printerName: printer_name, ...payload } = printJobOptions
 
-      expect(create).toHaveBeenCalledWith({ data: createPrintJobJson })
-      expect(response).toEqual(expectedResponse)
+      expect(create).toBeCalledWith({
+        data: { ...payload, printer_name, label_template_name: defaultState.tubeLabelTemplateName },
+      })
+      expect(success).toBeTruthy()
+      expect(message).toEqual('Barcode(s) successfully printed')
     })
 
-    it('unsuccessfully failed', async () => {
-      const failedResponse = {
-        success: false,
-        data: {
-          errors: [{ source: { pointer: 'api/labels' }, detail: 'failed' }],
-        },
-      }
-      create.mockReturnValue(failedResponse)
-
-      const expectedResponse = {
-        success: false,
-        data: {
-          message: 'api/labels failed',
+    //TODO: Check errors are correct
+    it('unsuccessful', async () => {
+      const create = vi.fn()
+      const rootState = { api: { printMyBarcode: { print_jobs: { create } } } }
+      const mockResponse = {
+        status: '422',
+        response: {
+          data: {
+            errors: [
+              {
+                source: {
+                  pointer: '/data/attributes/printer',
+                },
+                detail: "can't be blank",
+              },
+            ],
+          },
         },
       }
 
-      const response = await Actions.printJob({ getters }, params)
-
-      expect(create).toHaveBeenCalledWith({ data: createPrintJobJson })
-      expect(response).toEqual(expectedResponse)
-    })
-
-    it('unsuccessfully empty data', async () => {
-      const emptyResponse = { success: false, data: { errors: [] } }
+      create.mockRejectedValue(mockResponse)
 
-      create.mockReturnValue(emptyResponse)
-      const response = await Actions.printJob({ getters }, params)
+      // eslint-disable-next-line no-unused-vars
+      const { _, ...rest } = printJobOptions
 
-      const expectedResponse = {
-        success: false,
-        data: {
-          message: 'Unknown',
-        },
-      }
-
-      expect(create).toHaveBeenCalledWith({ data: createPrintJobJson })
-      expect(response).toEqual(expectedResponse)
-    })
-  })
-
-  describe('#createPrintJobJson', () => {
-    it('returns the correct json', () => {
-      const params = {
-        printerName: printerName,
-        barcodesList: barcodesList,
-        copies: copies,
-        suffix: suffix,
-      }
+      const { success, message } = await createPrintJob(
+        { rootState, state: defaultState },
+        { ...rest },
+      )
 
-      const result = Actions.createPrintJobJson(params, tubeLabelTemplateName)
-      expect(result).toEqual(createPrintJobJson)
+      expect(success).toBeFalsy()
+      // looks like there is an issue with the errors again. How many times do we have to fix this.
```

Comment:
What is wrong with it?

Reply:
I am just a bit unclear of why this is different to other API's. Print My Barcode is behaving differently I think?